File: app/src/main/java/io/github/sithengineer/motoqueiro/hardware/capture/LatLng.java
package io.github.sithengineer.motoqueiro.hardware.capture;

import android.location.Location;

public class LatLng {

  private final double lat;
  private final double lng;
  private final long timestamp;

  public LatLng(double lat, double lng, long timestamp) {
    this.lat = lat;
    this.lng = lng;
    this.timestamp = timestamp;
  }

  public LatLng(Location location) {
    this.lat = location.getLatitude();
    this.lng = location.getLongitude();
    this.timestamp = location.getTime();
  }

  public double getLat() {
    return lat;
  }

  public double getLng() {
    return lng;
  }

  public long getTimestamp() {
    return timestamp;
  }

  @Override public String toString() {
    return "lat=" + lat + ", lng=" + lng + ", timestamp=" + timestamp;
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/hardware/capture/LatLng.kt
package io.github.sithengineer.motoqueiro.hardware.capture

import android.location.Location

class LatLng {

  val lat: Double
  val lng: Double
  val timestamp: Long

  constructor(lat: Double, lng: Double, timestamp: Long) {
    this.lat = lat
    this.lng = lng
    this.timestamp = timestamp
  }

  constructor(location: Location) {
    this.lat = location.latitude
    this.lng = location.longitude
    this.timestamp = location.time
  }

  override fun toString(): String {
    return "lat=$lat, lng=$lng, timestamp=$timestamp"
  }
}
